From 97d873355e1bc5445a9b856a7d160600fd41e4c2 Mon Sep 17 00:00:00 2001
From: Ryan Houdek <Sonicadvance1@gmail.com>
Date: Thu, 3 Jul 2025 15:26:44 -0700
Subject: [PATCH] wineboot: Detect TSC frequency on Arm64.

Otherwise the `~MHz` registry key is initialized to the
maximum clock speed of the CPUs, which is known to break games.
---
 programs/wineboot/wineboot.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git b/programs/wineboot/wineboot.c a/programs/wineboot/wineboot.c
index 8359e5d4c44..02d192a1982 100644
--- b/programs/wineboot/wineboot.c
+++ a/programs/wineboot/wineboot.c
@@ -297,15 +297,6 @@ static UINT64 read_tsc_frequency(void)
     return freq;
 }
 
-#elif defined(__aarch64__)
-
-static UINT64 read_tsc_frequency(void)
-{
-    UINT64 tsc_frequency;
-    __asm__ volatile( "mrs %[Res], CNTFRQ_EL0" : [Res] "=r" (tsc_frequency) );
-    return tsc_frequency;
-}
-
 #else
 
 static UINT64 read_tsc_frequency(void)
-- 
2.50.0

